src/ImageSharp/Formats/Jpeg/JpegEncoderCore.cs

```diff
@@ -557,19 +584,10 @@ private void WriteStartOfFrame(int width, int height, int componentCount)
         /// <summary>
         /// Writes the StartOfScan marker.
         /// </summary>
-        /// <typeparam name="TPixel">The pixel format.</typeparam>
-        /// <param name="image">The pixel accessor providing access to the image pixels.</param>
         /// <param name="componentCount">The number of components in a pixel.</param>
-        /// <param name="cancellationToken">The token to monitor for cancellation.</param>
-        private void WriteStartOfScan<TPixel>(Image<TPixel> image, int componentCount, CancellationToken cancellationToken)
-            where TPixel : unmanaged, IPixel<TPixel>
+        /// <param name="componentIds">The componentId's.</param>
+        private void WriteStartOfScan(int componentCount, ReadOnlySpan<byte> componentIds)
         {
-            Span<byte> componentId = stackalloc byte[]
-            {
-                0x01,
-                0x02,
-                0x03
-            };
             Span<byte> huffmanId = stackalloc byte[]
```

Comment:
This could use the same ROSpan<byte> compiler optimization.
```suggestion
            // This uses a C#'s compiler optimization that refers to the static data segment of the assembly,
            // and doesn't incur any allocation at all.
            ReadOnlySpan<byte> huffmanId = new byte[]
```

Reply:
BT @gfoidl Love these optimization tips, keep them coming! 👍

Reply:
changed with 66a9d1ae